lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -65,13 +65,13 @@ def test_missing_keys(self, client, server_config, keys, user_ok, pbench_token):
         specified.
        """
         self.required_keys = [
-            p[0]
-            for p in DatasetsDetail(
+            key
+            for key, parameter in DatasetsDetail(
                 client.config, client.logger
             ).schema.parameters.items()
-            if p[1].required
+            if parameter.required
         ]
-        super().test_missing_keys(client, server_config, keys, user_ok, pbench_token)
+        super().missing_keys(client, server_config, keys, user_ok, pbench_token)
```

Comment:
The code is identical except the subclass instance that it uses. for example here its using `DatasetsDetail` class but for controller_list test it would use `ControllersList` class instance so that's why I kept it in the subclass.

Reply:
Thanks -- I didn't pick up on that detail -- I figured I must be missing something.

If you're up for it, you could try passing in the class (e.g., `DatasetsDetail`), either as an argument to the helper function (in lieu of `keys`) or as an attribute of the test superclass (and an argument to `_setup()`) which would enable getting rid of the subclass functions altogether, which would be cool.

Reply:
hmm that's good idea I will try that

Reply:
Let's move this conversation to [here](https://github.com/distributed-system-analysis/pbench/pull/2413#discussion_r686989507) in #2413.